Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/aspect/EventAspectAdvice.java

```diff
@@ -176,52 +178,15 @@ public void logFailure(JoinPoint joinPoint, Throwable e) {
             builder.setThrowable(e);
             builder.setInvoker(sig.getDeclaringType().toString());
             builder.setMethod(sig.getName());
+            builder.setService(service);
+            builder.setVersion(version);
 
+            builder.buildMessageID();
+            builder.buildTransactionID();
+            builder.buildDescription();
 
-            EventDirector director = new EventDirector();
```

Comment:
I'm not familiar with EventDirector.  Is there a reason the processing done by it is removed?  Is it used elsewhere or just here and is therefore not needed and should be deleted?

Reply:
EventDirector is a wrapper, which takes in a builder and then calls the methods         builder.createNewEvent(), builder.buildMessageID(), builder.buildTransactionID(), and builder.buildDescription() which I am calling manually

Essentially all it does is delegate calls on the builder itself and houses no logic of its own. I don't see the need to even use it if we have the builder right there to get the event from directly. EventDirector could have been an entirely static class or a simple helper class - I don't understand the design choice behind it.

Reply:
Got it.  Not disagreeing with what you're saying, just curious if EventDirector is used anywhere else.  If not, can we just delete it?

Reply:
The Before and After event builders use them in BaseEventAdviceDelegate.java... Getting rid of the Director class would require the same approach there, as I did here. Not that much effort if we did want to get rid it.